pkg/graveler/combined_iterator.go

```diff
@@ -122,3 +123,37 @@ func (c *CombinedIterator) Close() {
 	c.iterA.Close()
 	c.iterB.Close()
 }
+
+// FilterTombstoneIterator wraps a value iterator and filters out tombstones.
+type FilterTombstoneIterator struct {
+	iter ValueIterator
+}
+
```

Comment:
Consider just retype ValueIterator - it will have the same implementation, just without the 'iter' as a member

Reply:
synced f2f - cannot do :)